### DISCUSSION TITLE
Unable to specify port values for data links with application.properties or environment variables

### DISCUSSION THREAD

**Discussion by jbell-astrobotic** [yamcs/yamcs#839] [General]
Hi! I'm following the documentation for configuration properties here: https://docs.yamcs.org/yamcs-server-manual/administration/configuration/

The problem arising is that the port number does not seem to resolve properly to an integer. I've tried configuring this with the quickstart provided, but I continue to get the following error:
yamcs.myproject.yaml: dataLinks[1]: port is of type STRING, but should be INTEGER instead

I wanted to start a discussion first to see if others have ran into this issue and have resolved it before I submit a bug ticket.

**Comment by fqqb** [comment]
Correct, this should be fixed. The issue occurs during yaml validation which will wrongly treat all property values as strings.

The same issue would occur with floats and booleans.

Some parts of the Yamcs configuration are not (currently) validated, and so there everything would work fine, despite this bug.

**Reply by fqqb** [reply]
Made an issue here: https://github.com/yamcs/yamcs/issues/840
